docs/source/index.mdx

```diff
@@ -0,0 +1,85 @@
+# Filesystem
+
+🤗 Filesystem is a package that provides a pythonic file interface ([fsspec-compatible](https://filesystem-spec.readthedocs.io/en/latest/)) to the [Hugging Face Hub](https://huggingface.co/). It builds on top of [Hugging Face Hub client library](https://huggingface.co/docs/huggingface_hub/index) to interact with the Hub.
+
+## Installation
+
+```bash
+pip install hffs
+```
+
+## Usage
+
+`HfFileSystem` is the cental class of the library that holds connection information and allows typical filesystem style operations like `cp`, `mv`, `ls`, `du`, `glob`, `get_file`, `put_file` etc.
+
+```python
+>>> from hffs import HfFileSystem
+>>> fs = HfFileSystem("my-username/my-dataset-repo", repo_type="dataset")
+
+>>> # List files in a directory
+>>> fs.ls("data")
+['train.csv', 'test.csv']
+
+>>> # List all ".csv" files in a repo
+>>> fs.glob("**.csv")
+['data/train.csv', 'data/test.csv']
+
+>>> # Read the contents of a remote file 
+>>> with fs.open("data/train.csv", "r") as f:
+...     train_data = f.readlines()
+
+>>> # Read all the contents of a remote file at once as a string
+>>> train_data = fs.read_text("data/train.csv")
+
+>>> # Write a remote file
+>>> with fs.open("data/validation.csv", "w") as f:
+...     f.write("text,label")
+...     f.write("Fantastic movie!,good")
+```
+
+If not explicitly provided, the `repo_type` argument defaults to `model`. Besides `model`, other supported values are `dataset` and `space`.
+
+The optional `revision` argument can be passed to open a filesystem from a specific commit (any revision such as a branch or a tag name or a commit hash).
+
+Note that, unlike the built-in `open`, the mode in `fsspec`'s `open` defaults to binary mode, i.e. "rb". This means you must explicitly set encoding as "r"/"w" in case of text mode.
+
+## Integration
```

Comment:
Maybe it'd make more sense to move the Integration section before the Usage section? It might be good for the user to check if they can use a URL with an integration before they start using the filesystem operations.

Reply:
This order comes from the [`s3fs`](https://s3fs.readthedocs.io/en/latest/) docs, so I think I'll leave it as-is.